db/change_cache.go

```diff
@@ -184,29 +183,34 @@ func (c *changeCache) Init(dbcontext *DatabaseContext, notifyChange func(base.Se
 
 	heap.Init(&c.pendingLogs)
 
-	// background tasks that perform housekeeping duties on the cache
-	bgt, err := NewBackgroundTask("InsertPendingEntries", c.context.Name, c.InsertPendingEntries, c.options.CachePendingSeqMaxWait/2, c.terminator)
-	if err != nil {
-		return err
-	}
-	c.backgroundTasks = append(c.backgroundTasks, bgt)
-
-	bgt, err = NewBackgroundTask("CleanSkippedSequenceQueue", c.context.Name, c.CleanSkippedSequenceQueue, c.options.CacheSkippedSeqMaxWait/2, c.terminator)
-	if err != nil {
-		return err
-	}
-	c.backgroundTasks = append(c.backgroundTasks, bgt)
+	go func() {
+		dcpStartedErr := <-dcpStarted
```

Comment:
As mentioned, let's discuss this change, as I'm not clear on what problem it's solving yet that the original lock isn't.

Reply:
This resolves the problem of mutation events occurring while `DCPClient.Start()` is happening, because that's blocking in `DatabaseContext.NewDatabaseContext`. Due to locking in the `changeCache`, we can deadlock here in `DCPClient.Start` which will send blocking events which are locked in `changeCache.processEntry`. We don't want housekeeping tasks to also lock in `changeCache.processEntry` while we are opening the database context, so we wait the database context is actually created.

This is a summary of an voice chat.